viewing_party/party.py

```diff
@@ -1,23 +1,225 @@
 # ------------- WAVE 1 --------------------
 
-def create_movie(title, genre, rating):
-    pass
+from ast import If
+from types import NoneType
+from tests.test_constants import MOVIE_TITLE_1, USER_DATA_2
+
+
+def create_movie(movie_title, genre, rating):
+
+
+    movie = {}
+
+    movie["title"] = movie_title
+    movie["genre"] = genre
+    movie["rating"] = rating
+    
+    for item in movie.values():
+        if item == None:
+            return None
+    else:
+        return movie
+
+def add_to_watched(user_data, movie):
+
+    user_data = {
+    "watched": []
+    }   
+    copy = movie.copy()
+
+    user_data["watched"].append(copy)
+    
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data = {
+        "watchlist": []
+    }
+    copy1 = movie.copy()
+    
+    user_data["watchlist"].append(copy1)
+
+    return user_data
+
+def watch_movie(user_data, title):
+
+
+
+    for movie in user_data["watchlist"]:
+        if title == movie['title']:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+    return user_data
+            
+
+        
+        
+
+
+
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    
+    sum = 0
+    count = 0
+    for movie in user_data["watched"]:
+        sum += movie["rating"]
+        count +=1
+    if sum != 0:
+        avg = sum / count
+    else:
+        avg = 0.0
+
+    return avg
+
+def get_most_watched_genre(user_data):
+# will come back to this one to handle all test cases
+    for movie in user_data["watched"]:
+        if movie["genre"] == movie["genre"]:
+            return movie["genre"]
+        elif len(user_data["watched"]) == 0:
+            return None

```

Comment:
Hey Brooke, revisit this and I'll provide feedback 😄

Reply:
Approaches we discussed:
- Creating a counter based off of the genres and whichever has the higher count return that
- Creating a counter and a `most_freq` with each iteration increase count and most_freq
- (BEST approach, audrey highly suggests) Use a frequency map! Create a dictionary with the genres as a key and the values as their 'count' aka frequency.